<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>974</width>
    <height>702</height>
   </rect>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>theSlate</string>
  </property>
  <property name="windowIcon">
   <iconset theme="theslate" resource="icons.qrc">
    <normaloff>:/icons/icon.svg</normaloff>:/icons/icon.svg</iconset>
  </property>
  <property name="documentMode">
   <bool>true</bool>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QFrame" name="projectFrame">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="editorFrame">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QFrame" name="tabFrame">
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="tabButtons">
            <property name="spacing">
             <number>0</number>
            </property>
           </layout>
          </item>
          <item>
           <widget class="QPushButton" name="closeButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset theme="tab-close" resource="icons.qrc">
              <normaloff>:/icons/tab-close.svg</normaloff>:/icons/tab-close.svg</iconset>
            </property>
            <property name="flat">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QStackedWidget" name="tabs"/>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>974</width>
     <height>22</height>
    </rect>
   </property>
   <property name="nativeMenuBar">
    <bool>true</bool>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <widget class="QMenu" name="menuOpenFrom">
     <property name="title">
      <string>Open From</string>
     </property>
    </widget>
    <widget class="QMenu" name="menuOpen_Recent">
     <property name="title">
      <string>Open Recent</string>
     </property>
     <property name="icon">
      <iconset theme="document-open-recent" resource="icons.qrc">
       <normaloff>:/icons/document-open-recent.svg</normaloff>:/icons/document-open-recent.svg</iconset>
     </property>
    </widget>
    <widget class="QMenu" name="menuReload_Using_Encoding">
     <property name="title">
      <string>Reload Using Encoding</string>
     </property>
    </widget>
    <widget class="QMenu" name="menuChange_File_Encoding">
     <property name="title">
      <string>Change File Encoding</string>
     </property>
    </widget>
    <addaction name="actionNew"/>
    <addaction name="separator"/>
    <addaction name="actionOpen"/>
    <addaction name="menuOpenFrom"/>
    <addaction name="menuOpen_Recent"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="actionSave_All"/>
    <addaction name="menuChange_File_Encoding"/>
    <addaction name="separator"/>
    <addaction name="actionRevert"/>
    <addaction name="menuReload_Using_Encoding"/>
    <addaction name="actionClose"/>
    <addaction name="separator"/>
    <addaction name="actionPrint"/>
    <addaction name="separator"/>
    <addaction name="actionSettings"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <widget class="QMenu" name="menuPaste_from_Clipboard_History">
     <property name="title">
      <string>Paste from Clipboard History</string>
     </property>
     <property name="icon">
      <iconset theme="edit-paste" resource="icons.qrc">
       <normaloff>:/icons/edit-paste.svg</normaloff>:/icons/edit-paste.svg</iconset>
     </property>
    </widget>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="actionCut"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="menuPaste_from_Clipboard_History"/>
    <addaction name="separator"/>
    <addaction name="actionSelect_All"/>
    <addaction name="separator"/>
    <addaction name="actionFind_and_Replace"/>
    <addaction name="separator"/>
    <addaction name="actionComment"/>
    <addaction name="actionUncomment"/>
   </widget>
   <widget class="QMenu" name="menuCode">
    <property name="title">
     <string>&amp;Code</string>
    </property>
    <property name="icon">
     <iconset theme="commit" resource="icons.qrc">
      <normaloff>:/icons/commit.svg</normaloff>:/icons/commit.svg</iconset>
    </property>
    <addaction name="actionNo_Highlighting"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <property name="icon">
     <iconset theme="help-contents" resource="icons.qrc">
      <normaloff>:/icons/help-contents.svg</normaloff>:/icons/help-contents.svg</iconset>
    </property>
    <addaction name="actionFile_Bug"/>
    <addaction name="actionSources"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuSource_Control">
    <property name="title">
     <string>So&amp;urce Control</string>
    </property>
    <addaction name="actionPush"/>
    <addaction name="actionPull"/>
    <addaction name="separator"/>
    <addaction name="actionFetch"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>Wi&amp;ndow</string>
    </property>
    <addaction name="actionUse_Menubar"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="actionNew_Window"/>
    <addaction name="separator"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuCode"/>
   <addaction name="menuSource_Control"/>
   <addaction name="menuWindow"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="iconSize">
    <size>
     <width>24</width>
     <height>24</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
  </widget>
  <widget class="QDockWidget" name="filesDock">
   <property name="windowTitle">
    <string>Files</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QTreeView" name="projectTree">
       <property name="contextMenuPolicy">
        <enum>Qt::CustomContextMenu</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="sourceControlDock">
   <property name="windowTitle">
    <string>Sou&amp;rce Control</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="GitWidget" name="gitWidget">
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_7">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
    </layout>
   </widget>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset theme="document-new" resource="icons.qrc">
     <normaloff>:/icons/document-new.svg</normaloff>:/icons/document-new.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="toolTip">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset theme="document-save" resource="icons.qrc">
     <normaloff>:/icons/document-save.svg</normaloff>:/icons/document-save.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="toolTip">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset theme="document-open" resource="icons.qrc">
     <normaloff>:/icons/document-open.svg</normaloff>:/icons/document-open.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Open</string>
   </property>
   <property name="toolTip">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset theme="application-exit" resource="icons.qrc">
     <normaloff>:/icons/application-exit.svg</normaloff>:/icons/application-exit.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
  </action>
  <action name="actionCopy">
   <property name="icon">
    <iconset theme="edit-copy" resource="icons.qrc">
     <normaloff>:/icons/edit-copy.svg</normaloff>:/icons/edit-copy.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Copy</string>
   </property>
   <property name="toolTip">
    <string>Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="icon">
    <iconset theme="edit-cut" resource="icons.qrc">
     <normaloff>:/icons/edit-cut.svg</normaloff>:/icons/edit-cut.svg</iconset>
   </property>
   <property name="text">
    <string>C&amp;ut</string>
   </property>
   <property name="toolTip">
    <string>Cut</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="icon">
    <iconset theme="edit-paste" resource="icons.qrc">
     <normaloff>:/icons/edit-paste.svg</normaloff>:/icons/edit-paste.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Paste</string>
   </property>
   <property name="toolTip">
    <string>Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionNo_Highlighting">
   <property name="text">
    <string>&amp;No Highlighting</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset theme="help-about" resource="icons.qrc">
     <normaloff>:/icons/help-about.svg</normaloff>:/icons/help-about.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;About</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutRole</enum>
   </property>
  </action>
  <action name="actionPrint">
   <property name="icon">
    <iconset theme="document-print" resource="icons.qrc">
     <normaloff>:/icons/document-print.svg</normaloff>:/icons/document-print.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Print</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionPush">
   <property name="text">
    <string>&amp;Push</string>
   </property>
  </action>
  <action name="actionPull">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>P&amp;ull</string>
   </property>
  </action>
  <action name="actionFetch">
   <property name="text">
    <string>&amp;Fetch</string>
   </property>
  </action>
  <action name="actionStart">
   <property name="icon">
    <iconset theme="system-run">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Start</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionSave_All">
   <property name="icon">
    <iconset theme="document-save">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Sa&amp;ve All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Alt+S</string>
   </property>
  </action>
  <action name="actionContinue">
   <property name="icon">
    <iconset theme="media-playback-start">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Continue</string>
   </property>
  </action>
  <action name="actionStep_Into">
   <property name="icon">
    <iconset theme="debug-step-into">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Step Into</string>
   </property>
  </action>
  <action name="actionStep_Over">
   <property name="icon">
    <iconset theme="debug-step-over">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Step Over</string>
   </property>
  </action>
  <action name="actionStep_Out">
   <property name="icon">
    <iconset theme="debug-step-out">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Step Out</string>
   </property>
  </action>
  <action name="actionPause">
   <property name="icon">
    <iconset theme="media-playback-pause">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Pause</string>
   </property>
  </action>
  <action name="actionFind_and_Replace">
   <property name="icon">
    <iconset theme="edit-find-replace" resource="icons.qrc">
     <normaloff>:/icons/edit-find-replace.svg</normaloff>:/icons/edit-find-replace.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Find and Replace...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="icon">
    <iconset theme="document-save-as">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Save As...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionRevert">
   <property name="icon">
    <iconset theme="view-refresh" resource="icons.qrc">
     <normaloff>:/icons/view-refresh.svg</normaloff>:/icons/view-refresh.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Revert</string>
   </property>
  </action>
  <action name="actionFile_Bug">
   <property name="icon">
    <iconset theme="tools-report-bug" resource="icons.qrc">
     <normaloff>:/icons/tools-report-bug.svg</normaloff>:/icons/tools-report-bug.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;File Bug</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionSources">
   <property name="icon">
    <iconset theme="commit" resource="icons.qrc">
     <normaloff>:/icons/commit.svg</normaloff>:/icons/commit.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Sources</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset theme="edit-undo" resource="icons.qrc">
     <normaloff>:/icons/edit-undo.svg</normaloff>:/icons/edit-undo.svg</iconset>
   </property>
   <property name="text">
    <string>U&amp;ndo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="icon">
    <iconset theme="edit-redo" resource="icons.qrc">
     <normaloff>:/icons/edit-redo.svg</normaloff>:/icons/edit-redo.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="icon">
    <iconset theme="configure" resource="icons.qrc">
     <normaloff>:/icons/configure.svg</normaloff>:/icons/configure.svg</iconset>
   </property>
   <property name="text">
    <string>Sett&amp;ings</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+,</string>
   </property>
   <property name="menuRole">
    <enum>QAction::PreferencesRole</enum>
   </property>
  </action>
  <action name="actionClose">
   <property name="icon">
    <iconset theme="document-close" resource="icons.qrc">
     <normaloff>:/icons/document-close.svg</normaloff>:/icons/document-close.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;Close</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="actionNew_Window">
   <property name="icon">
    <iconset theme="window-new" resource="icons.qrc">
     <normaloff>:/icons/window-new.svg</normaloff>:/icons/window-new.svg</iconset>
   </property>
   <property name="text">
    <string>&amp;New Window</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+N</string>
   </property>
  </action>
  <action name="actionSelect_All">
   <property name="icon">
    <iconset theme="edit-select-all" resource="icons.qrc">
     <normaloff>:/icons/edit-select-all.svg</normaloff>:/icons/edit-select-all.svg</iconset>
   </property>
   <property name="text">
    <string>Select All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionUse_Menubar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Use Menubar</string>
   </property>
  </action>
  <action name="actionComment">
   <property name="text">
    <string>Comment</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionUncomment">
   <property name="text">
    <string>Uncomment</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+D</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GitWidget</class>
   <extends>QWidget</extends>
   <header>SourceControl/gitwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections/>
</ui>
